### PR TITLE
Added toVoid, <String>isEmpty and isEqualTo functions

### PR DESCRIPTION
**Pull request by pawel-schmidt**

### CODE REVIEW

File: rx-extensions/src/main/java/com/appunite/rx/functions/Functions1.java [hunk 1]
```diff
@@ -129,4 +142,38 @@ public String call(CharSequence charSequence) {
             }
         };
     }
+
+    @Nonnull
+    public static Func1<CharSequence, Boolean> isEmpty() {
```
Comment (jacek-marchwicki) [inline rx-extensions/src/main/java/com/appunite/rx/functions/Functions1.java]: maybe isEmptyOrNull?

File: rx-extensions/src/main/java/com/appunite/rx/functions/Functions1.java [hunk 2]
```diff
@@ -129,4 +142,38 @@ public String call(CharSequence charSequence) {
...
+        return new Func1<CharSequence, Boolean>() {
+            @Override
+            public Boolean call(CharSequence charSequence) {
+                return charSequence.length() == 0;
```
Comment (jacek-marchwicki) [inline rx-extensions/src/main/java/com/appunite/rx/functions/Functions1.java]: please add some tests

File: rx-extensions/src/main/java/com/appunite/rx/functions/Functions1.java [hunk 3]
```diff
@@ -129,4 +142,38 @@ public String call(CharSequence charSequence) {
...
+    }
+
+    /**
+     * Checks if propagated value is equal to any of passed values.
+     * @param values values which are compared to propagated value.
+     * @return true if propagated value is equal to at least one of passed values, false otherwise.
+     * If values length is equal to 0 or passed null reference this method returns false.
+     */
+    @Nonnull
+    public static Func1<? super Object, Boolean> isEqualTo(@Nullable final Object... values) {
```
Comment (jacek-marchwicki) [inline rx-extensions/src/main/java/com/appunite/rx/functions/Functions1.java]: Is it array usefull?
I suggest to add @Nonnull
checkArgument(values != null, "Values can not be null");
checkArgument(values.length > 0, "You need to speciffy at least one object");

File: rx-extensions/src/main/java/com/appunite/rx/functions/Functions1.java (hunk 1 above)
Comment (pawel-schmidt) [inline rx-extensions/src/main/java/com/appunite/rx/functions/Functions1.java]: Prefer opposite order, so changing to `isNullOrEmpty`.

File: rx-extensions/src/main/java/com/appunite/rx/functions/Functions1.java (hunk 3 above)
Comment (pawelschmidt) [inline rx-extensions/src/main/java/com/appunite/rx/functions/Functions1.java]: It makes sense. This project has no Guava dependencies. I'm going to add simple class `Preconditions` with `checkNotNull` and `checkState` implementation.

Comment (pawelschmidt) [inline rx-extensions/src/main/java/com/appunite/rx/functions/Functions1.java]: I've noticed there is `internal.Preconditions` :)

File: rx-extensions/src/main/java/com/appunite/rx/functions/Functions1.java (hunk 2 above)
Comment (pawelschmidt) [inline rx-extensions/src/main/java/com/appunite/rx/functions/Functions1.java]: Can you point me an example where function is tested?

File: rx-extensions/src/main/java/com/appunite/rx/functions/Functions1.java [hunk 4]
```diff
@@ -129,4 +154,28 @@ public String call(CharSequence charSequence) {
...
+     */
+    @Nonnull
+    public static Func1<? super Object, Boolean> isEqualTo(@Nonnull final Object... values) {
+        checkNotNull(values, "Values cannot be null");
+        checkState(values.length > 0, "You need to specify at least one object.");
+        return new Func1<Object, Boolean>() {
+            @Override
+            public Boolean call(final Object o) {
+                for (final Object value : values) {
+                    if (Objects.equals(o, value)) {
```
Comment (jacek-marchwicki) [inline rx-extensions/src/main/java/com/appunite/rx/functions/Functions1.java]: you can't use this method... Objects.equals is not supported on older android versions.. you need to use Objects.equal from guava .. I think is backported in internal

**Comment by jacek-marchwicki** [comment]
:-1:

**Comment by pawel-schmidt** [comment]
@jacek-marchwicki :point_up:

**Comment by jacek-marchwicki** [comment]
lgtm

**Pull request merged**